src/domains/customer.js

```diff
@@ -1,26 +1,62 @@
-const prisma = require('../utils/prisma')
+const { customer } = require('../utils/prisma')
 
-/**
- * This will create a Customer AND create a new Contact, then automatically relate them with each other
- * @tutorial https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#create-a-related-record
- */
-const createCustomerDb = async (name, phone, email) => await prisma.customer.create({
-  data: {
-    name,
-    contact: {
-      create: {
-        phone,
-        email
+const getCustomerByIdDb = async (customerId) => {
+  const foundCustomer = await customer.findFirst({
+    where: {
+      id: Number(customerId)
+    }
+  })
+
+  return foundCustomer
+}
+
+const createCustomerDb = async (name, phone, email) => {
+  const createdCustomer = await customer.create({
+    data: {
+      name,
+      contact: {
+        create: {
+          phone,
+          email
+        }
       }
+    },
+    include: {
+      contact: true
+    }
+  })
+
+  return createdCustomer
+}
+
+const updateCustomerByIdDb = async (fields, customerId) => {
+  const updatedCustomer = await customer.update({
+    where: {
+      id: Number(customerId)
+    },
+    data: {
+      name: fields.name,
+      ...(fields.contact && {

```

Comment:
I've not seen `...` used like that before. What is it doing there?

Reply:
Using this three dots you can open obj or array in your code for example in my case I should have data: {} instead this if I will not use three dots I'll get this: {data:{}} and the app will give me error after that because here is syntax error. So you can use it for copy everything what inside obj or array and also in the same time open that. Also you can use it like this:
```
const val1 = [1,2,3,4,5,6]
const val2 = [...val1]
```
And in this case val2 will be the same one that val 2.